packages/atomic/src/components/commerce/result-template-components/product-template-decorators.tsx

```diff
@@ -0,0 +1,42 @@
+import {Product} from '@coveo/headless/commerce';
+import {
+  InteractiveItemContext,
+  InteractiveItemContextEvent,
+  ItemContext,
+  ItemContextEvent,
+  itemContext,
+} from '../../common/item-list/item-decorators';
+
+/**
+ * A [StencilJS property decorator](https://stenciljs.com/) to be used for product template components.
+ * This allows the Stencil component to fetch the current product from its rendered parent, the `atomic-product` component.
+ *
+ * Example:
+ * @ProductContext() private product!: Product;
```

Comment:
```suggestion
  * ```
  * @ProductContext() private product!: Product;
  * ```
```